### PR TITLE
Misc Cleanup 3

### PR DESCRIPTION
**Pull request by engineer124**
This particular cleanup focuses on enums and structs. The biggest change here is reenabling the warning for having a trailing comma on an enum, which would then be enforced by Jenkins. But, we can keep it disabled if people prefer that instead.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Ani/z_en_ani.h [hunk 1]
```diff
@@ -28,10 +28,10 @@ typedef struct EnAni {
     /* 0x2FC */ EnAniActionFunc actionFunc;
 } EnAni; // size = 0x300
 
-enum EnAniType {
-  /* else */ ANI_TYPE_STANDING,         // unfinshed and unused
-  /* 1    */ ANI_TYPE_TREE_HANGING = 1,
-};
+typedef enum EnAniType {
+  /* 0 */ ANI_TYPE_STANDING, // unfinshed and unused
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Ani/z_en_ani.h:32]: Not really fond of this, sure else isn't standard but its never specified zero anywhere, in the code its always the else clause.

File: src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.h [hunk 2]
```diff
@@ -55,14 +55,14 @@ typedef struct EnBigpo {
 
 // well ver is regular, dampe basement ver is summoned
 // on spawn, 3/possible fires are turned into chosenfire
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.h:57]:
```suggestion
// well version is "regular" (spawns automatically), dampe basement version is "summoned"
// on scene enter, 3 "possiblefire" are turned into "chosenfire" at random
```
Not the scope of the PR but this documentation should be more specific

File: src/overlays/actors/ovl_En_Nwc/z_en_nwc.c [hunk 3]
```diff
@@ -32,14 +32,14 @@ void EnNwc_DrawAdultBody(Actor* thisx, PlayState* play);
 s32 EnNwc_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx);
 EnHs* EnNwc_FindGrog(PlayState* play);
 
-enum EnNiwState {
+typedef enum EnNiwState {
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Nwc/z_en_nwc.c:35]:
```suggestion
typedef enum EnNwcState {
```

File: src/overlays/actors/ovl_En_Nwc/z_en_nwc.c [hunk 4]
```diff
@@ -32,14 +32,14 @@ void EnNwc_DrawAdultBody(Actor* thisx, PlayState* play);
...
     /* -1 */ NWC_STATE_NIW_LOADED = -1,  // set after loading object_niw
     /*  0 */ NWC_STATE_CHECK_BREMAN = 0, // checking for breman mask
     /*  1 */ NWC_STATE_TURNING,          // turning to face a new direction to explore
     /*  2 */ NWC_STATE_HOPPING_FORWARD,  // hopping to go explore
     /*  3 */ NWC_STATE_FOLLOWING,        // following the player
-    /*  4 */ NWC_STATE_RUNNING,          // running from the player after failed breman march
-};
+    /*  4 */ NWC_STATE_RUNNING           // running from the player after failed breman march
+} EnNiwState;
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Nwc/z_en_nwc.c:42]:
```suggestion
} EnNwcState;
```

File: src/overlays/actors/ovl_En_Ani/z_en_ani.h (hunk 1 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Ani/z_en_ani.h:32]: Ah, I see what it was going for. Just by including it here though, it's still being forced to take a value by having this enum though. Perhaps the comment can be expanded upon instead explaining this "else" behavior?

**Review by tom-overton — COMMENTED** [review]
Do we want to decide on whether enums should be anonymous or not in this PR? Or should that wait for a follow-up?

I noticed the inconsistency because the enum is not anonymous in `z_bombjima.c` but it *is* anonymous in `z_bombjimb.h` directly below it.

**Comment by engineer124** [comment]
> Do we want to decide on whether enums should be anonymous or not in this PR? Or should that wait for a follow-up?

I was considering it yes, this PR would be a good time to make it consistent if we wanted to. I don't have a preference either way though

**Comment by tom-overton** [comment]
> I was considering it yes, this PR would be a good time to make it consistent if we wanted to. I don't have a preference either way though

I took a look and there's a slight preference towards anonymous enums in the codebase, but it's not a huge skew in any direction. It makes me think this is probably something we can revisit in a future PR, since it's not as if there's a clear preference that everyone has expressed, and only some files are different. We should probably discuss which people prefer before changing something like 500 enums.

File: src/overlays/actors/ovl_En_Sth/z_en_sth.h [hunk 5]
```diff
@@ -16,7 +16,7 @@ typedef enum {
     /* 2 */ STH_TYPE_SWAMP_SPIDER_HOUSE_CURED, // cursed is EnSsh
     /* 3 */ STH_TYPE_MOON_LOOKING, // South Clock Town, looking up at the moon
     /* 4 */ STH_TYPE_OCEANSIDE_SPIDER_HOUSE_GREET, // looking for shelter
-    /* 5 */ STH_TYPE_OCEANSIDE_SPIDER_HOUSE_PANIC, // shelter was not enough
+    /* 5 */ STH_TYPE_OCEANSIDE_SPIDER_HOUSE_PANIC // shelter was not enough
     // Other values: Actor will spawn and animate with arm waving, no further interaction.
 } EnSthTypes;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.h:21]:
```suggestion
} EnSthType;
```

File: src/overlays/actors/ovl_En_Stone_heishi/z_en_stone_heishi.c [hunk 6]
```diff
@@ -100,7 +100,7 @@ typedef enum {
     /* 3 */ EN_STONE_HEISHI_ANIM_WAVE,
     /* 4 */ EN_STONE_HEISHI_ANIM_SIT_AND_REACH,
     /* 5 */ EN_STONE_HEISHI_ANIM_DRINK_2,
-    /* 6 */ EN_STONE_HEISHI_ANIM_STAND_UP,
+    /* 6 */ EN_STONE_HEISHI_ANIM_STAND_UP
 } EnStoneHeishiAnimations;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Stone_heishi/z_en_stone_heishi.c:104]:
```suggestion
} EnStoneHeishiAnimation;
```

File: src/overlays/actors/ovl_En_Tanron4/z_en_tanron4.c [hunk 7]
```diff
@@ -23,12 +23,12 @@ void EnTanron4_FlyNearActor(EnTanron4* this, PlayState* play);
...
     /* 0 */ SEAGULL_FLY_FLAP,
-    /* 1 */ SEAGULL_FLY_GLIDE,
+    /* 1 */ SEAGULL_FLY_GLIDE
 } SeagullFlyState;
 
 typedef enum {
     /* 0 */ SEAGULL_TIMER_FLY_STATE,
-    /* 1 */ SEAGULL_TIMER_CHOOSE_TARGET,
+    /* 1 */ SEAGULL_TIMER_CHOOSE_TARGET
 } SeagullTimers;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Tanron4/z_en_tanron4.c:32]:
```suggestion
} SeagullTimer;
```

File: src/overlays/actors/ovl_En_Skb/z_en_skb.h [hunk 8]
```diff
@@ -10,7 +10,7 @@ typedef void (*EnSkbActionFunc)(struct EnSkb*, PlayState*);
 #define ENSKB_GET_F0(thisx) (((thisx)->params >> 4) & 0xF)
 
 typedef enum {
-    /* 0 */ ENSKB_PARAMS_0,
+    /* 0 */ ENSKB_PARAMS_0
 } EnSkbParams;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Skb/z_en_skb.h:14]:
```suggestion
    /* 0 */ ENSKB_PARAM_0
} EnSkbParam;
```

File: src/overlays/actors/ovl_Item_B_Heart/z_item_b_heart.h [hunk 9]
```diff
@@ -4,8 +4,8 @@
 #include "global.h"
 
 typedef enum {
-    /*  0 */ BHEART_PARAM_NORMAL,     // Spawn Parameter for Normal Scale.
-    /* 35 */ BHEART_PARAM_SMALL = 35, // Spawn Parameter for Small Scale.
+    /*  0 */ BHEART_PARAM_NORMAL,    // Spawn Parameter for Normal Scale.
+    /* 35 */ BHEART_PARAM_SMALL = 35 // Spawn Parameter for Small Scale.
 } ItemBHeartParams;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Item_B_Heart/z_item_b_heart.h:9]:
```suggestion
} ItemBHeartParam;
```

File: src/overlays/actors/ovl_Mir_Ray/z_mir_ray.h [hunk 10]
```diff
@@ -19,7 +19,7 @@ typedef enum {
     /* 0x7 */ MIRRAY_SPIRIT_TOPROOM_COBRA1,
     /* 0x8 */ MIRRAY_SPIRIT_TOPROOM_COBRA2,
     /* 0x9 */ MIRRAY_GANONSCASTLE_SPIRITTRIAL_DOWNLIGHT,
-    /* 0xA */ MIRRAY_MAX,
+    /* 0xA */ MIRRAY_MAX
 } MirRayBeamLocations;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Mir_Ray/z_mir_ray.h:23]:
```suggestion
} MirRayBeamLocation;
```

File: src/overlays/actors/ovl_Obj_Ocarinalift/z_obj_ocarinalift.h [hunk 11]
```diff
@@ -25,8 +25,8 @@ typedef struct ObjOcarinalift {
 } ObjOcarinalift; // size = 0x178
 
 typedef enum ObjOcarinaliftParams {
-    /* 0x0 */ OBJOCARINALIFT_PARAMSC_0,
-    /* 0x1 */ OBJOCARINALIFT_PARAMSC_1
+    /* 0 */ OBJOCARINALIFT_PARAMSC_0,
+    /* 1 */ OBJOCARINALIFT_PARAMSC_1
 } ObjOcarinaliftParams;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Ocarinalift/z_obj_ocarinalift.h:30]:
```suggestion
} ObjOcarinaliftParam;
```
Can you remove the `s` from the other parts as well?

File: include/z64.h [hunk 12]
```diff
@@ -681,15 +681,15 @@ enum fram_command {
...
-};
-
-enum fram_mode {
-    FRAM_MODE_NOP = 0,
-    FRAM_MODE_ERASE,
-    FRAM_MODE_WRITE,
-    FRAM_MODE_READ,
-    FRAM_MODE_STATUS
-};
+} fram_command;
```
Comment (hensldm) [inline include/z64.h:684]:
```suggestion
} FramCommand;
```
To match current style.

File: include/z64.h [hunk 13]
```diff
@@ -681,15 +681,15 @@ enum fram_command {
...
+
+typedef enum fram_mode {
+    /* 0 */ FRAM_MODE_NOP,
+    /* 1 */ FRAM_MODE_ERASE,
+    /* 2 */ FRAM_MODE_WRITE,
+    /* 3 */ FRAM_MODE_READ,
+    /* 4 */ FRAM_MODE_STATUS
+} fram_mode;
```
Comment (hensldm) [inline include/z64.h:692]:
```suggestion
} FramMode;
```

**Review by hensldm — COMMENTED** [review]
Some thoughts. Mainly around removing 's` (i.e. having non-plural enum names).

File: src/overlays/actors/ovl_En_Ani/z_en_ani.h (hunk 1 above)
Comment (isghj5) [reply src/overlays/actors/ovl_En_Ani/z_en_ani.h]: What do you think of moving ANI_TYPE_STANDING to below TREE_HANGING? I feel like else condition should come after actually used values, and TREE_HANGING already has a hard value.

Comment (engineer124) [reply src/overlays/actors/ovl_En_Ani/z_en_ani.h]: Yeah, fine with me.

**Review by isghj5 — APPROVED** [review]

**Review by jpburnett — COMMENTED** [review]
I approve

**Pull request merged**